crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -328,7 +362,54 @@ mod tests {
 
     use super::*;
     use crate::flox::tests::flox_instance;
-    use crate::prelude::Channel;
+
+    #[tokio::test]
+    async fn adds_substituter_metadata() {
+        let (flox, _temp_dir_handle) = flox_instance();
+
+        let publish_flake_ref: PublishFlakeRef = "git+ssh://git@github.com/flox/dummy"
+            .parse::<FlakeRef>()
+            .unwrap()
```

Comment:
You can use [and_then](https://doc.rust-lang.org/std/result/enum.Result.html#method.and_then) here to avoid multiple `unwrap`s.

Reply:
I'm getting complaints about Error types when I do that, maybe because `parse::<FlakeRef>` and `try_into()` have different error types? Is there a way to do it I'm missing?

Reply:
Sorry, lost track of this. This is a matter of style so if it's a blocker feel free to ignore it. Generally you would call `.map_err` to convert one error into another if possible.

Reply:
Oh I think this was outdated anyways, sorry